src/plugins/add-cal/add-calendar.js

```diff
@@ -0,0 +1,155 @@
+					}
+				};
+
+			// Initiate dictionary
+			i18nDict = i18nDict[ $( "html" ).attr( "lang" ) || "en" ];
+			i18nDict = {
+				addto: i18nDict[ "addcal-addto" ],
+				calendar: i18nDict[ "addcal-calendar" ],
+				ical: i18nDict[ "addcal-ical" ]
+			};
+
+			//Set date stamp with the date modified
+			if ( $( "time[property='dateModified'][datetime]" ).length ) {
+				event_details.dtStamp = new Date( $( "time[property='dateModified']" ).attr( "datetime" ) );
+			} else {
+				event_details.dtStamp = new Date( $( "time[property~='dateModified']" ).text() );
+			}
+
+			event_details.dtStamp = event_details.dtStamp.toISOString().replace( /\..*[0-9]/g, "" ).replace( /-|:|\./g, "" );
+
+			i_len = properties.length;
+			for ( i = 0; i < i_len; i++ ) {
+				prop_cache = properties[ i ];
+				switch ( prop_cache.getAttribute( "property" ) ) {
+				case "name":
+					if ( $( prop_cache ).parentsUntil( ( "." + componentName ), "[typeof=Place]" ).length ) {
+						event_details.place = prop_cache.textContent;
+					} else {
+						event_details.name = prop_cache.textContent;
+					}
+					break;
+				case "description":
+					event_details.description = prop_cache.textContent.replace( /(\r\n|\n|\r)/gm, " " );
+					break;
+				case "startDate":
+					event_details.sDate = new Date( prop_cache.getAttribute( "datetime" ) ).toISOString().replace( /\..*[0-9]/g, "" ).replace( /-|:|\./g, "" );
+					break;
+				case "endDate":
+					event_details.eDate = new Date( prop_cache.getAttribute( "datetime" ) ).toISOString().replace( /\..*[0-9]/g, "" ).replace( /-|:|\./g, "" );
+					break;
+				case "location":
+					if ( !prop_cache.getAttribute( "typeof" ) ) {
+						event_details.place = prop_cache.textContent;
+					}
+					break;
+				case "streetAddress":
+					event_details.place += " " + prop_cache.textContent;
+					break;
+				case "addressLocality":
+					event_details.place += " " + prop_cache.textContent;
+					break;
+				case "addressRegion":
+					event_details.place += " " + prop_cache.textContent;
+					break;
+				case "postalCode":
+					event_details.place += " " + prop_cache.textContent;
+					break;
+				}
+			}
+
+			// Error handling
+			if ( !event_details.name ) {
+				throw componentName + ": Event title is missing.";
+			} else if ( !event_details.sDate ) {
+				throw componentName + ": Start date is missing.";
+			} else if ( !event_details.eDate ) {
+				throw componentName + ": End date is missing.";
+			}
+
+			//Set Unique Identifier (UID) and Date Stamp (DSTAMP)
+			event_details.uid = window.location.href.replace( /\.|-|\/|:|[G-Zg-z]/g, "" ).toUpperCase().substr( -10 ) + "-" + event_details.sDate + "-" + event_details.dtStamp;
+
+			googleLink = encodeURI( "https://www.google.com/calendar/render?action=TEMPLATE" +  "&text=" + event_details.name +  "&details=" + event_details.description +  "&dates=" + event_details.sDate + "/" + event_details.eDate + "&location=" + event_details.place );
```

Comment:
Have you fixed the issue you had when the event description/name contain an apostrophe `'` character?

Reply:
Yes resolved at line 134